apps/dsb-message-broker/src/auth/login.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { LoginReturnDataDTO } from './login-return-data.dto';
+import { ethers } from 'ethers';
+
+@Injectable()
+export class LoginService {
+    constructor(private readonly configService: ConfigService) {}
+
+    public async mbIdentifiers(jwtToken?: string): Promise<Partial<LoginReturnDataDTO>> {
+        const jwtEncodedPayload = jwtToken.split('.')[1];
+        const jwtDecodedPayload = ethers.utils.base64.decode(jwtEncodedPayload);
+        const jwtPayload = JSON.parse(String.fromCharCode.apply(null, jwtDecodedPayload));
```

Comment:
I would change fromCharCode decoding function to https://docs.ethers.io/v5/api/utils/strings/#utils-toUtf8String since the output of base64.decode is utf8 like bytes while fromCharCode is designed to work with utf16 afair.

Reply:
replaced with ethers.utils.toUtf8String